prowler/providers/aws/services/waf/waf_service.py

```diff
@@ -94,6 +107,123 @@ def _list_resources_for_web_acl(self, regional_client):
+        try:
+            for rule in regional_client.list_rules().get("Rules", []):
+                arn = f"arn:aws:waf-regional:{regional_client.region}:{self.audited_account}:rule/{rule['RuleId']}"
+                self.rules[arn] = Rule(
+                    arn=arn,
+                    region=regional_client.region,
+                    id=rule.get("RuleId", ""),
+                    name=rule.get("Name", ""),
+                )
+
+        except Exception as error:
+            logger.error(
+                f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+
+    def _get_rule(self, rule):
+        logger.info(f"WAFRegional - Getting Rule {rule.name}...")
+        try:
+            get_rule = self.regional_clients[rule.region].get_rule(RuleId=rule.id)
+            for predicate in get_rule.get("Rule", {}).get("Predicates", []):
+                rule.predicates.append(
+                    Predicate(
+                        negated=predicate.get("Negated", False),
+                        data_id=predicate.get("DataId", ""),
+                    )
+                )
+
+        except KeyError:
+            logger.error(f"Rule {rule.name} not found in {rule.region}.")
+
+    def _list_rule_groups(self, regional_client):
+        logger.info("WAFRegional - Listing Regional Rule Groups...")
+        try:
+            for rule_group in regional_client.list_rule_groups().get("RuleGroups", []):
+                arn = f"arn:aws:waf-regional:{regional_client.region}:{self.audited_account}:rulegroup/{rule_group['RuleGroupId']}"
+                self.rule_groups[arn] = RuleGroup(
+                    arn=arn,
+                    region=regional_client.region,
+                    id=rule_group.get("RuleGroupId", ""),
+                    name=rule_group.get("Name", ""),
+                )
+
+        except Exception as error:
+            logger.error(
+                f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+
+    def _list_activated_rules_in_rule_group(self, rule_group):
+        logger.info(
+            f"WAFRegional - Listing activated rules in Rule Group {rule_group.name}..."
+        )
+        try:
+            for rule in (
+                self.regional_clients[rule_group.region]
+                .list_activated_rules_in_rule_group(RuleGroupId=rule_group.id)
+                .get("ActivatedRules", [])
+            ):
+                rule_group.rules.append(ACLRule(id=rule.get("RuleId", "")))
+
+        except Exception as error:
+            logger.error(
+                f"{rule_group.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+
+
+class Predicate(BaseModel):
+    """Predicate Model for WAF and WAFRegional"""
+
+    negated: bool
+    data_id: str
+
+
+class ACLRule(BaseModel):
+    """ACL Rule Model for WAF and WAFRegional"""
+
+    id: str
+
```

Comment:
Please, don't use this reference and access to the rules dictionary when needed.